libminifi/src/io/DescriptorStream.cpp

```diff
@@ -84,11 +84,10 @@ int DescriptorStream::readData(uint8_t *buf, int buflen) {
   if (!IsNullOrEmpty(buf)) {
     auto size_read = ::read(fd_, buf, buflen);
 
-    if (size_read != buflen) {
+    if (size_read < 0) {
```

Comment:
Yes.
DescriptorStream is only used for server socket callback registration. 
Server socket callback is only used in ControllerSocketProtocol.
ControllerSocketProtocol only reads UTF strings, which works correctly with this change.

Reply:
Hmm, I know customers are using this functionality successfully -- but I also know something broke along the way ( haven't had time to investigate ). What was the chief problem that you saw? Is this a necessary part of this PR? If not can we improve testing on the piece it visits?

Reply:
Testing doesn't work at all without it.
So UTF strings (which works) is a special way of communication as the length of the string proceeds the string itself, you always know how many bytes to read.
However in general case when you try to read all the data from the socket, no matter how much is that, you fail. 
The use case is creating a big buffer and read whatever read we have in the socket to it.
In this case I would expect the API to tell me how many bytes did it manage to read. Instead it returned -1 in case it couldn't completely fill the buffer. 
The only way to deal with this would had been reading bytes on by one, but that still wouldn't work as the call after reading the last byte would end up blocking...